milli/src/search/matching_words.rs

```diff
@@ -0,0 +1,204 @@
+use std::collections::HashSet;
+use std::cmp::{min, Reverse};
+use std::collections::BTreeMap;
+use std::ops::{Index, IndexMut};
+
+use levenshtein_automata::{DFA, Distance};
+
+use crate::search::query_tree::{Operation, Query};
+
+use super::build_dfa;
+
+type IsPrefix = bool;
+
+/// The query tree builder is the interface to build a query tree.
+#[derive(Default)]
+pub struct MatchingWords {
+    dfas: Vec<(DFA, String, u8, IsPrefix)>,
+}
+
+impl MatchingWords {
+    /// Lists all words which can be considered as a match for the query tree.
+    pub fn from_query_tree(tree: &Operation) -> Self {
+        let mut dfas: Vec<_> = fetch_queries(tree)
+            .into_iter()
+            .map(|(w, t, p)| (build_dfa(w, t, p), w.to_string(), t, p))
+            .collect();
+        dfas.sort_unstable_by_key(|(_dfa, query_word, _typo, _is_prefix)| Reverse(query_word.len()));
```

Comment:
Could you add a comment on top of this line explaining the reason why we reverse sort them by length, please?